app/views/api/v2/models/main.json.rabl

```diff
@@ -3,3 +3,10 @@ object @model
 extends "api/v2/models/base"
 
 attributes :info, :created_at, :updated_at, :vendor_class, :hardware_model
+
+if params.has_key?(:show_permissions)
+  node(:can_edit) { |model| model.authorized?(:edit_model)}
```

Comment:
I think this causes many queries, could we reuse Authorizer in here to cache results for the whole collection? see app/services/authorizer.rb or let me know if you need a hand. Most of the time, users have same permissions for all resources of the same type so it's better to cache the collections.